flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/optimization/RegularizationType.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.optimization
+
+import breeze.numerics._
+import org.apache.flink.ml.math.{BLAS, Vector}
+import org.apache.flink.ml.math.Breeze._
+import breeze.linalg.{norm => BreezeNorm, Vector => BreezeVector, max}
+
+abstract class RegularizationType extends Serializable{
+  /** Calculates and applies the regularization amount and the regularization parameter
+    *
+    * @param weights The old weights
+    * @param effectiveStepSize The effective step size for this iteration
+    * @param regParameter The current regularization parameter
+    * @return A tuple whose first element is the updated weight vector and the second is the
+    *         new regularization parameter
+    */
+  def applyRegularization(weights: Vector, effectiveStepSize: Double, regParameter: Double):
+  (Vector, Double)
```

Comment:
We are not currently using the regularization value anywhere, but it could be useful to keep a history of it.